frontend/src/pages/groups/GroupCreatePage.tsx

```diff
@@ -0,0 +1,66 @@
+import { FunctionComponent } from 'react'
+import LayoutWithNav from '../../layouts/LayoutWithNav'
+import { Group, GroupInput } from '../../types/api-types'
+import { gql, useMutation } from '@apollo/client'
+import { useHistory } from 'react-router-dom'
+import GroupForm from './GroupForm'
+
+const ADD_GROUP = gql`
+  mutation Groups($input: GroupInput!) {
+    addGroup(input: $input) {
+      id
+    }
+  }
+`
+
+const GroupCreatePage: FunctionComponent = () => {
+  const history = useHistory()
+
+  const [
+    addGroup,
+    { loading: mutationIsLoading, error: mutationError },
+  ] = useMutation<{ addGroup: Group }>(ADD_GROUP)
+
+  const onSubmit = (input: GroupInput) => {
+    // Create the group and then redirect to its view/edit page
+    addGroup({ variables: { input } })
+      .then(({ data }) => {
+        if (data) {
+          const newGroupId = data.addGroup.id
+          history.push(`/group/${newGroupId}`)
+        }
+      })
+      .catch((error) => {
+        console.log(error)
```

Comment:
Suggestion (blocking): `console.error` for the error

Reply:
Question: what's the difference between this catch and the `mutationError` variable?

Reply:
The `error` from the catch block has the type `any`, and it's a different shape from the `mutationError`, which is fully-typed.

I'm using both error variables for different purposes: the `catch` block is used to print the error to the console, while the `mutationError` is displayed to the user.